x-pack/plugins/synthetics/server/synthetics_service/formatters/public_formatters/tcp.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ConfigKey, TCPFields } from '../../../../common/runtime_types';
+import {
+  objectToJsonFormatter,
+  stringToJsonFormatter,
+} from '../private_formatters/formatting_utils';
+import { Formatter, commonFormatters } from './common';
+import { tlsFormatters } from './tls';
+
+export type TCPFormatMap = Record<keyof TCPFields, Formatter>;
+export const tcpFormatters: TCPFormatMap = {
+  ...tlsFormatters,
+  ...commonFormatters,
+  [ConfigKey.METADATA]: objectToJsonFormatter,
```

Comment:
We shouldn't be using to `ToJSON` formatter here. this should likely be `objectFormatter` not `objectToJSONFormatter`
```suggestion
  [ConfigKey.METADATA]: objectFormatter,
```